streaming/src/main/scala/org/apache/spark/streaming/dstream/InputDStream.scala

```diff
@@ -47,6 +49,23 @@ abstract class InputDStream[T: ClassTag] (@transient ssc_ : StreamingContext)
   /** This is an unique identifier for the input stream. */
   val id = ssc.getNewInputStreamId()
 
+  /**
+   * A rate estimator configured by the user to compute a dynamic ingestion bound for this stream.
+   * @see `RateEstimator`
+   */
+  protected [streaming] val rateEstimator = ssc.conf
+                                               .getOption("spark.streaming.RateEstimator")
+                                               .getOrElse("noop") match {
```

Comment:
I'd break this logic into its own method. The list of implementations might grow.